### PR TITLE
[ENG-2961] Fix eslint errors for web extension

### PR DESCRIPTION
**Pull request by dhriaznov**
# 🔘 PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no API changes)

# 📜 Background
We'd like to resolve the existing eslint errors.

Issue Link: #[[ENG-2961](https://linear.app/xverseapp/issue/ENG-2961/fix-eslint-errors-for-web-extension)]
Context Link (if applicable): https://secretkeylabs.slack.com/archives/C05QCJM9D1N/p1695912058419199?thread_ts=1695912058.419199&cid=C05QCJM9D1N

# 🔄 Changes
Enumerate the changes made in this pull request, detailing what has been modified, added, or removed. Include technical details and implications if necessary.
- Fixed the eslint errors
- Made some code refactoring and removal of the unused code

Impact:
- This PR doesn't contain any dramatic changes, though it touches the transactions logic (both incoming and regular ones)

# 🖼 Screenshot / 📹 Video
Include screenshots or a video demonstrating the changes. This is especially helpful for UI changes.

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: src/app/hooks/queries/useAppConfig.ts [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
+      try {
+        const response = await getAppConfig();
+        if (response.data.btcApiURL && network.type === 'Mainnet' && !btcApiUrl) {
+          const updatedNetwork = { ...network, btcApiUrl: response.data.btcApiURL };
+          dispatch(ChangeNetworkAction(updatedNetwork, networkAddress, ''));
+        }
+        return response;
+      } catch (err) {
+        return Promise.reject(err);
       }
```
Comment (victorkirov) [inline src/app/hooks/queries/useAppConfig.ts:23]:
```suggestion
      const response = await getAppConfig();
      if (response.data.btcApiURL && network.type === 'Mainnet' && !btcApiUrl) {
        const updatedNetwork = { ...network, btcApiUrl: response.data.btcApiURL };
        dispatch(ChangeNetworkAction(updatedNetwork, networkAddress, ''));
      }
      return response;

```

**Review by victorkirov — CHANGES_REQUESTED** [review]
This is awesome 🎉  Just left 2 comments for cleanup 🙏

**Comment by dhriaznov** [comment]
Thanks @victorkirov, I updated the PR, could you take a second look, please?

**Review by victorkirov — APPROVED** [review]
LGTM 🚀 🚢

**Comment by blehitsahmad** [comment]
@dhriaznov good to merge -- [comment](https://linear.app/xverseapp/issue/ENG-2961/fix-eslint-errors-for-web-extension#comment-a755e2a3) added to the linear ticket.

**Pull request merged**